portmapper/proxy.go

```diff
@@ -65,10 +66,23 @@ func (p *proxyCommand) Start() error {
 
 func (p *proxyCommand) Stop() error {
 	if p.cmd.Process != nil {
-		if err := p.cmd.Process.Signal(os.Interrupt); err != nil {
+		if err := p.cmd.Process.Signal(syscall.SIGTERM); err != nil {
 			return err
 		}
-		return p.cmd.Wait()
+
+		waitChan := make(chan error)
+		go func() {
+			waitChan <- p.cmd.Wait()
+		}()
+
+		select {
+		case result := <-waitChan:
+			return result
+		case <-time.After(15 * time.Second):
```

Comment:
This leads to a stray timer. It is better to create a timer and properly stop it.
```go
t := time.NewTimer(15 * time.Second)
defer t.Stop()
select {
	case result := <-waitChan:
		return result
	case <- t.C:
		if err := p.cmd.Process.Signal(os.Kill); err != nil {
			return err
}
```